### ISSUE TITLE
Make `idx/Query` Serialisable 

### ISSUE THREAD

**Issue by prateek** [m3db/m3ninx#42]
Add the following methods to `idx/query.go`:

```
func Marshall(Query) ([]byte, error) { ... }
func Unmarshall([]byte) (Query, error) { ... }
```

Can eliminate `Query.SearchQuery()` after that.

**Comment by prateek** [comment]
Also would be good to expose methods that allowed the returning pooled return types.

**Comment by jeromefroe** [comment]
Closed by #56
